extrinsic.go

```diff
@@ -160,21 +182,70 @@ func (e *ExtrinsicDecoder) Process() {
 	}
 
 	if e.ContainsTransaction {
-		result["account_id"] = e.Address
-		result["signature"] = e.Signature
-		result["nonce"] = e.Nonce
-		result["era"] = e.Era
-		result["extrinsic_hash"] = e.ExtrinsicHash
+		result.AccountId = e.Address
+		result.Signature = e.Signature
+		result.Nonce = e.Nonce
+		result.Era = e.Era
+		result.ExtrinsicHash = e.ExtrinsicHash
+	}
+	result.CallCode = e.CallIndex
+	result.CallModuleFunction = call.Call.Name
+	result.CallModule = call.Module.Name
+	result.Params = e.Params
+	e.Value = &result
```

Comment:
It is better to return the map[string]interface, so that the versions are compatible

Reply:
In addition, a method of ``map[string]interface`` to ``GenericExtrinsic`` is needed, so that Encode can be called directly with